test/EndToEndTests/Common/Microsoft.OData.Client.E2E.TestCommon/Common/HttpClientResponseMessage.cs

```diff
@@ -0,0 +1,134 @@
+
+        public HttpClientResponseMessage(HttpResponseMessage response)
+        {
+            _response = response;
+        }
+
+        public string? GetHeader(string headerName)
+        {
+            if (_response.Headers.TryGetValues(headerName, out var values))
+            {
+                return string.Join(",", values);
+            }
+
+            if (_response.Content.Headers.TryGetValues(headerName, out values))
+            {
+                return string.Join(",", values);
+            }
+
+            return null;
+        }
+
+        public async Task<Stream> GetStreamAsync()
+        {
+            return await _response.Content.ReadAsStreamAsync();
+        }
+
+        public IEnumerable<KeyValuePair<string, string>> Headers
+        {
+            get
+            {
+                foreach (var header in _response.Headers)
+                {
+                    yield return new KeyValuePair<string, string>(header.Key, string.Join(",", header.Value));
+                }
+
+                foreach (var contentHeader in _response.Content.Headers)
+                {
+                    yield return new KeyValuePair<string, string>(contentHeader.Key, string.Join(",", contentHeader.Value));
+                }
+            }
+        }
+
+        public void SetHeader(string headerName, string headerValue)
+        {
+            // HttpResponseMessage doesn't allow directly setting headers on the response, 
+            throw new NotSupportedException("Setting headers on HttpResponseMessage is not supported.");
+        }
+
+        public int StatusCode
+        {
+            get
+            {
+                return (int)_response.StatusCode;
+            }
+            set
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        public string StatusDescription
+        {
+            get => _response.ReasonPhrase;
+            set
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        public Stream GetStream()
+        {
+            return _response.Content.ReadAsStream();
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _response?.Dispose();
+            }
+
+            _disposed = true;
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore();
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual async ValueTask DisposeAsyncCore()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            _response?.Dispose();
+            _disposed = true;
+
+            await ValueTask.CompletedTask;
+        }

```

Comment:
```suggestion
        protected virtual ValueTask DisposeAsyncCore()
        {
            if (_disposed)
            {
                return ValueTask.CompletedTask;
            }

            _response?.Dispose();
            _disposed = true;

            return ValueTask.CompletedTask;
        }
```

Reply:
I've done away with the dispose logic in this class.